src/lib/hal/services/HalClient.ts

```diff
@@ -35,7 +35,8 @@ export interface HalClient {
     params: any,
     data: any,
     resourceConstructor: HalResourceConstructor<T>,
-    method: HttpMethod.POST | HttpMethod.PATCH
+    method: HttpMethod.POST | HttpMethod.PATCH | HttpMethod.PUT,
+    returnError: boolean
```

Comment:
The point is that the data in the error is meaningful. When the data is put into the HttpError, it isn't parsed as an object, it's just put into the message string. In the case of edition schedule, the return value is a status object containing an array of warnings/errors, each with codes, messages and overlaps of their own.

Reply:
This could be modelled using an Exception, the validation errors that DC throw all follow the same pattern --> https://amplience.com/docs/api/dynamic-content/management/index.html#section/Usage/Error-Response-Format

Reply:
I could also remove this functionality for now and just leave it up to the user to parse/interpret whatever's in `HttpError.response.data.errors[...]` with the interfaces left exported. That might be a better way to get these changes in without any larger scale changes to the error exception type. What do you think?

Reply:
Is the `HttpError` from underlying Axios lib? If so, I think we should create our own HTTP Error Exception class for this.

Responses & Exceptions form part of the 'API contract' SDK & their users, we should try to minimise the use 3rd party libs from these 'contracts'.

Reply:
The HttpError is provided by us, but it's safe to assume that people have built tools catching these exceptions already, so any changes should be built on top of these error types if possible. I'm just asking for now though, if it would make sense for me to just leave this change out and parse the errors directly from the data in my pending changes on the `dc-cli` side.